### PR TITLE
Convert to standard Go benchmarks

### PR DESCRIPTION
**Pull request by moorereason**
Reference discussion in #370

### CODE REVIEW

File: CONTRIBUTE.adoc [hunk 1]
```diff
@@ -29,3 +29,47 @@ Don’t get discouraged if you don't get an immediate response, this is a side-p
...
+
+Since we use Ginkgo, it's generally a good idea to skip the normal tests when running benchmarks.
+To do that, use something like `go test -run=XXX` to skip normal tests (assuming no test contains "XXX").
+
+It's also a good idea to run multiple iterations of the benchmarks to avoid anomolies and gather better statistics.
+
+Example command:
+
+```
+go test -run=XXX -bench=.* -benchmem -count=10 ./... | tee bench-branch1.txt
```
Comment (xcoulon) [inline CONTRIBUTE.adoc]: the `-bench=.*` flag gives me the following error:
```
zsh: no matches found: -bench=*
```
but `-bench=.` works, instead.

Comment (xcoulon) [inline CONTRIBUTE.adoc]:
```s
go test -run=XXX -bench=. -benchmem -count=10 ./... | tee bench-branch1.txt
```

File: CONTRIBUTE.adoc [hunk 2]
```diff
@@ -29,3 +29,47 @@ Don’t get discouraged if you don't get an immediate response, this is a side-p
...
+```
+
+After making changes to your branch, run the same benchmarks and save to a second file:
+
+```
+go test -run=XXX -bench=.* -benchmem -count=10 ./... | tee bench-branch2.txt
```
Comment (xcoulon) [inline CONTRIBUTE.adoc]:
```s
go test -run=XXX -bench=. -benchmem -count=10 ./... | tee bench-branch2.txt
```

File: CONTRIBUTE.adoc [hunk 3]
```diff
@@ -29,3 +29,47 @@ Don’t get discouraged if you don't get an immediate response, this is a side-p
...
+```
+
+Compare results using [benchstat](https://godoc.org/golang.org/x/perf/cmd/benchstat).  Example output:
```
Comment (xcoulon) [inline CONTRIBUTE.adoc]:
```s
Compare results using https://godoc.org/golang.org/x/perf/cmd/benchstat[benchstat].  Example output:
```

Comment (xcoulon) [reply CONTRIBUTE.adoc]: This is an asciidoc file, not a markdown one ;)

**Comment by moorereason** [comment]
Fixed the issues you raised and rebased against master due to merge conflicts.

File: CONTRIBUTE.adoc [hunk 4]
```diff
@@ -32,7 +32,7 @@ Testing existing features is a good way to contribute to the project, too!
 
 == Benchmarking
 
-This project uses [Ginkgo](http://onsi.github.io/ginkgo/) as the main testing framework but uses the standard Go toolchain for benchmarking and performance comparisons.
+This project uses http://onsi.github.io/ginkgo/[Ginkgo] as the main testing framework but uses the standard Go toolchain for benchmarking and performance comparisons.
```
Comment (xcoulon) [inline CONTRIBUTE.adoc]: ah good catch, I believe I missed that one earlier ;)

**Comment by xcoulon** [comment]
thanks a lot @moorereason! 🎉

**Pull request merged**